lib/foreman_maintain/utils/sync_plans.rb

```diff
@@ -0,0 +1,112 @@
+module ForemanMaintain
+  module Utils
+    module SyncPlans
+      include ForemanMaintain::Concerns::SystemHelpers
+      include ForemanMaintain::Concerns::HammerHelpers
+
+      FILE_TO_STORE_MANIPULATED_DATA = 'data.yml'.freeze
+      MAKE_SYNC_PLANS_DISABLE = 'add'.freeze
+      MAKE_SYNC_PLANS_ENABLE = 'remove'.freeze
+
+      class << self
+        def ids(db_feature, enabled = true)
+          enabled = enabled ? 't' : 'f'
+          db_feature.query(
+            <<-SQL
+              SELECT id FROM katello_sync_plans WHERE enabled ='#{enabled}'
+            SQL
+          ).map { |r| r['id'].to_i }
+        end
+
+        def count_of_disabled_plans
+          record_ids_needs_to_enable.length
+        end
+
+        def record_ids_needs_to_enable
+          file_path = File.expand_path FILE_TO_STORE_MANIPULATED_DATA
+          existing_data = fetch_yaml_data(file_path)
+          return [] if update_infokeys_absent?(existing_data)
+          existing_data['sync_plans']['update']['ids'].to_a
+        end
+
+        def make_disable(ids)
+          update_records(ids, false)
+        end
+
+        def make_enable
+          ids = record_ids_needs_to_enable
+          update_records(ids, true, MAKE_SYNC_PLANS_ENABLE)
+        end
+
+        private
+
+        def update_records(ids, enabled, sync_plans_action = MAKE_SYNC_PLANS_DISABLE)
+          updated_record_ids = []
+          ids.each do |sp_id|
+            updated_record_ids = execute_update_cmd(
+              sp_id, enabled, updated_record_ids, sync_plans_action
+            )
+          end
+          store_to_yaml(updated_record_ids, sync_plans_action)
+        end
+
+        def execute_update_cmd(id, enabled, updated_record_ids, sync_plans_action)
+          cmd = "#{hammer_command} sync-plan update --id #{id} --enabled #{enabled}"
+          result = execute(cmd)
+          if result.include?('Sync plan updated')
+            updated_record_ids << id
+          else
+            store_to_yaml(updated_record_ids, sync_plans_action)
+            raise result
+          end
+          updated_record_ids
+        end
+
+        def store_to_yaml(ids, action_name)
+          file_path = File.expand_path FILE_TO_STORE_MANIPULATED_DATA
```

Comment:
@iNecas, 
I have other query about argument for ForemanMaintain.storage.
Here argument would be a label like in metadata we have label or it will represent class name.